README.md

```diff
@@ -9,67 +9,204 @@ Ceph-related Docker files.
+docker build -t <daemon tag> daemon/
+```
+
+## Contribution workflow
+The goal when adding contributions should be to make modifications or additions to the least
+specific files in order to reuse as much code as possible. Only specify specific changes when they
+absolutely apply only to the specific flavor(s) and not to others.
+
+### Fixing a bug
+1. Stage the flavor on which the bug was found (`make FLAVORS_TO_BUILD=<bugged build> stage`).
+2. Use the `find-src` script or `files-sources` list to locate the bugged file's source location.
+3. Edit the source location to fix the bug.
+4. Build test versions of the images (`make FLAVORS_TO_BUILD=<bugged build> build`).
+5. Test the images you built in your environment.
+6. Make a PR of your changes.
+
+### Adding a feature
+1. Determine the scope of the feature
+   - To which Ceph versions should the feature be added?
+   - To which distros should the feature be added?
+   - As a general guideline, new features should usually be added to all Ceph versions and distros.
+2. Add relevant changes to files in the project structure such that they will be added to only the
+   Ceph versions and distros in scope for the feature.
+3. Build test versions of the images (`make FLAVORS_TO_BUILD=<in-scope-flavors> build`).
+4. Test the images in your environment.
+5. Make a PR of your changes.
+
+### Adding a Ceph release
+Ideally, adding a new Ceph release is fairly easy. In the best case, all that needs done is adding
+flavors for the new Ceph version to the Makefile. At minimum, `ALL_BUILDABLE_FLAVORS` must be
+updated in the Makefile.
+
+In the worst case, trying to make as few modifications as possible:
+1. Add flavors for new Ceph versions to the Makefile.
+   - At minimum: `ALL_BUILDABLE_FLAVORS`.
+2. Edit `src/` files to support the new version if necessary, making sure not to break previous
+   versions.
+3. Edit `src/ALL/<distro>` files to support the new version if necessary, making sure not to break
+   previous versions.
+4. Add `src/<new ceph version>` files to support the new version if necessary.
+5. Build test versions of the images (`make FLAVORS_TO_BUILD=<new release flavors> build`).
+6. Test the images in your environment.
+7. Make a PR of your changes.
+
+### Adding a distro build
+1. Add flavors for the new distro to the Makefile.
+   - At minimum: `ALL_BUILDABLE_FLAVORS`.
+2. Add a `src/ALL/<new distro>` directory for the new distro
+   - Make sure to install all the required packages for `daemon-base` (see
+     `src/daemon-base/__CEPH_BASE_PACKAGES__)` and for `daemon` (see
+     `src/daemon/__DAEMON_PACKAGES__`).
+   - Make sure to specify all `__VAR__` files without sane defaults for the container builds.
+   - Make sure to override any `__VAR__` file sane defaults that do not apply to the new distro.
+   - Refer to other distros for inspiration.
+3. If necessary (try to avoid this as much as possible), add a `src/<ceph release>/<new distro>`
```

Comment:
I think we have to be explicit why we shall avoid this.

Reply:
Yes.